pygfx/controls/_base.py

```diff
@@ -0,0 +1,46 @@
+from typing import Tuple
+
+from ..linalg import Vector3
+from ..utils.viewport import Viewport
+
+
+class Controller:
+    """Abstract base controller."""
+
+    def update_camera(self, camera: "Camera") -> "Controller":
+        """Update the transform of the camera with the internal transform."""
+        rot, pos, zoom = self.get_view()
+        camera.rotation.copy(rot)
+        camera.position.copy(pos)
+        camera.zoom = zoom
+        return self
+
+    def add_default_event_handlers(self, viewport, camera):
```

Comment:
can you type hint viewport to be `Union[Viewport, Renderer]` or something like that, to clearly signal the duck typing here?